inspire_json_merger/utils/utils.py

```diff
@@ -0,0 +1,89 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Helper functions for authors."""
+
+from __future__ import absolute_import, division, print_function
+
+import re
+
+split_on_re = re.compile('[\.\s-]')
+
+
+def scan_author_string_for_phrases(s):
+    """Scan a name string and output an object representing its structure.
+    Example:
+        Sample output for the name 'Jingleheimer Schmitt, John Jacob, XVI.' is::
+            {
+                'TOKEN_TAG_LIST' : ['lastnames', 'nonlastnames', 'titles', 'raw'],
+                'lastnames'      : ['Jingleheimer', 'Schmitt'],
+                'nonlastnames'   : ['John', 'Jacob'],
+                'titles'         : ['XVI.'],
+                'raw'            : 'Jingleheimer Schmitt, John Jacob, XVI.'
+            }
+    :param s: the input to be lexically tagged
+    :type s: string
+    :returns: dict of lexically tagged input items.
+    :rtype: dict
+    """
+
+    retval = {
+        'TOKEN_TAG_LIST': [
+            'lastnames',
+            'nonlastnames',
+            'titles',
+            'raw'],
+        'lastnames': [],
+        'nonlastnames': [],
+        'titles': [],
+        'raw': s}
+    l = s.split(',')
+    if len(l) < 2:
+        # No commas means a simple name
+        new = s.strip()
+        new = s.split(' ')
+        if len(new) == 1:
+            retval['lastnames'] = new        # rare single-name case
+        else:
+            retval['lastnames'] = new[-1:]
+            retval['nonlastnames'] = new[:-1]
+            for tag in ['lastnames', 'nonlastnames']:
+                retval[tag] = [x.strip() for x in retval[tag]]
+                retval[tag] = [re.split(split_on_re, x)
+                               for x in retval[tag]]
+                # flatten sublists
+                retval[tag] = [item for sublist in retval[tag]
+                               for item in sublist]
+                retval[tag] = [x for x in retval[tag] if x != '']
+    else:
+        # Handle lastname-first multiple-names case
+        retval['titles'] = l[2:]             # no titles? no problem
+        retval['nonlastnames'] = l[1]
+        retval['lastnames'] = l[0]
+        for tag in ['lastnames', 'nonlastnames']:
+            retval[tag] = retval[tag].strip()
+            retval[tag] = re.split(split_on_re, retval[tag])
+            # filter empty strings
+            retval[tag] = [x for x in retval[tag] if x != '']
+        retval['titles'] = [x.strip() for x in retval['titles'] if x != '']
+
+    return retval
```

Comment:
This function is working on a raw string rather than unicode objects. There is a risk it explodes on non-ascii characters. Also was it copied from json-merger? If yes, maybe you can refer to it and explain the difference.